blank/webpack.config.js

```diff
@@ -1,54 +1,73 @@
-const { CleanWebpackPlugin } = require('clean-webpack-plugin')
+const webpack = require('webpack')
 const CopyPlugin = require('copy-webpack-plugin')
-const HtmlWebpackPlugin = require('html-webpack-plugin')
 const TerserPlugin = require('terser-webpack-plugin')
+const HtmlWebpackPlugin = require('html-webpack-plugin')
+const { CleanWebpackPlugin } = require('clean-webpack-plugin')
+const ImageMinimizerPlugin = require('image-minimizer-webpack-plugin')
 
-const webpack = require('webpack')
-
-const root = path.resolve(__dirname, 'src')
-const botonicPath = path.resolve(__dirname, 'node_modules', '@botonic', 'react')
-
-const terserPlugin = new TerserPlugin({
-  parallel: true,
-  sourceMap: true,
-  terserOptions: {
-    keep_fnames: true,
-  },
-})
+const ROOT = path.resolve(__dirname, 'src')
+const OUTPUT_PATH = path.resolve(__dirname, 'dist')
+const BOTONIC_PATH = path.resolve(
+  __dirname,
+  'node_modules',
+  '@botonic',
+  'react'
+)
 
-const MODE_DEV = 'development'
-const MODE_PROD = 'production'
+const WEBPACK_MODE = {
+  DEVELOPMENT: 'development',
+  PRODUCTION: 'production',
+}
 
 const BOTONIC_TARGETS = {
+  ALL: 'all',
   DEV: 'dev',
   NODE: 'node',
   WEBVIEWS: 'webviews',
   WEBCHAT: 'webchat',
 }
 
+const WEBPACK_ENTRIES = {
+  DIR: 'webpack-entries',
+  DEV: 'dev-entry.js',
+  NODE: 'node-entry.js',
+  WEBCHAT: 'webchat-entry.js',
+  WEBVIEWS: 'webviews-entry.js',
+}
+
+const TEMPLATES = {
+  WEBCHAT: 'webchat.template.html',
+  WEBVIEWS: 'webview.template.html',
+}
+
+const UMD_LIBRARY_TARGET = 'umd'
+const BOTONIC_LIBRARY_NAME = 'Botonic'
+const WEBCHAT_FILENAME = 'webchat.botonic.js'
+
 function sourceMap(mode) {
-  // changing it from inline-source-map to cheap-eval-source-map, build time improved from 48s to 40s
-  if (mode === MODE_PROD) {
-    // Typescript: "inline-source-map" does not map Typescript correctly but there's a patch I didn't test https://github.com/webpack/webpack/issues/7172#issuecomment-414115819
-    // from https://webpack.js.org/configuration/devtool/ inline-source-map is slow and not good for production
-    //return 'cheap-eval-source-map'; // fast builds, not for production, transformed code (lines only)
-    // slow, good for production. A full SourceMap is emitted as a separate file. but doesn't add a reference comment to the bundle. Useful if you only want SourceMaps to map error stack traces from error reports, but don't want to expose your SourceMap for the browser development tools.
-    return 'hidden-source-map'
-  } else if (mode === MODE_DEV) {
-    // 'eval-source-map' would be a good fit for staging (slow but generates original code)
-    // from documentation: quick build time, very quick rebuild, transformed code (lines only)
-    return 'cheap-eval-source-map' //callstacks show links to TS code
-  } else {
+  if (mode === WEBPACK_MODE.PRODUCTION) return 'hidden-source-map'
+  else if (mode === WEBPACK_MODE.DEVELOPMENT) return 'eval-cheap-source-map'
```

Comment:
Is `eval-cheap-source-map` correct (new code) or it has to be `cheap-eval-source-map` (old code)? I guess it has been changed in webpack 5, but just to be sure it's not a mistake.

Reply:
yes, sure, it has been changed in webpack 5